invenio_banners/services/service.py

```diff
@@ -7,28 +7,61 @@
 
 """Banner Service API."""
 
-from invenio_access.permissions import system_identity
 from invenio_records_resources.services import RecordService
+from invenio_records_resources.services.base import LinksTemplate
 
-from invenio_banners.records.models import BannerModel
+from ..records.models import BannerModel
+from ..services.errors import BannerNotExistsError
 
 
 class BannerService(RecordService):
     """Banner Service."""
 
-    def read(self, url_path):
+    def read(self, identity, id):
         """Retrieve a banner."""
-        # TODO: not implemented/tested yet
-        # self.require_permission(identity, "read")
-        banner = BannerModel.get_active(url_path)
-        return self.result_item(self, system_identity, banner)
-
-    def search(self, identity, params=None, **kwargs):
-        """Search for resources matching the querystring."""
-        # TODO: not implemented/tested yet
-        # self.require_permission(identity, "read")
-        banners = BannerModel.search()
-        return self.result_list(self, system_identity, banners)
+        # resolve and require permission
+        self.require_permission(identity, "read")
+
+        banner = self.record_cls.get(id)
+
+        # check if banner exists
+        if banner is None:
+            raise BannerNotExistsError(id)
+
+        return self.result_item(
+            self,
+            identity,
+            banner,
+            links_tpl=self.links_item_tpl,
+        )
+
+    def read_active(self, identity, active, url_path, params=None):
```

Comment:
Shall we change it to plural given that it returns a list?
```suggestion
    def read_actives(self, identity, active, url_path, params=None):
```